addons/common/functions/fnc_registerItemReplacement.sqf

```diff
@@ -20,76 +20,87 @@
 params [["_oldItem", "", [0,""]], ["_newItems", "", ["", []]], ["_replaceInherited", false, [false]]];
 TRACE_3("registerItemReplacement",_oldItem,_newItems,_replaceInherited);
 
-// CBA player event handler function
-private _fnc_replaceItems = {
-    params ["_unit"];
 
-    private _items = items _unit;
-    if (_items isEqualTo GVAR(oldItems)) exitWith {};
+// Setup on first run
+if (isNil QGVAR(itemReplacements)) then {
 
-    private _newItems = _items - GVAR(oldItems);
-    _newItems = _newItems arrayIntersect _newItems; // Get unique items only
-    if (_newItems isEqualTo []) exitWith {
-        GVAR(oldItems) = _items;
-    };
+    // CBA player event handler function
+    DFUNC(registerItemReplacement_replaceItems) = compileFinal toString {
+        params ["_unit"];
```

Comment:
this function could also possibly be useful for running gear replacement on an AI
if someone had a reason to do so

Reply:
move this to a proper function?